### PR TITLE
Nine-block control & inspector strategies POC

### PR DESCRIPTION
**Pull request by bkrmendy**
08-31-7zbki-0nu5e

## Problem
The inspector in its current for can only manipulate properties on a one-by-one basis. The drawback of this approach is that there exist some interactions, where multiple props have to be adjusted (such as converting a container to/from flex or changing the flex direction). 

## Solution
To fix this, we're trying a new approach that is geared towards compositionality/customisability. In this new approach, an interaction through a given control is run through a cascade of inspector strategies (inspired by the canvas strategies system that we already have). These inspector strategies detect if they are applicable, and if they are, they return an array of editor actions that are dispatched as the result of that UI interaction.

## This PR
This PR is a PoC/first step in the approach outlined above. It introduces
- a "nine-block" control for setting `align-items` and `justify-content` on a flex container
- the inspector strategies system

### CODE REVIEW

File: editor/src/utils/feature-switches.ts [hunk 1]
```diff
@@ -27,6 +29,7 @@ let FeatureSwitches: { [feature in FeatureName]: boolean } = {
   'Re-parse Project Button': !(PRODUCTION_CONFIG as boolean),
   'Performance Test Triggers': !(PRODUCTION_CONFIG as boolean),
   'Canvas Strategies Debug Panel': false,
+  'Nine block control': true,
```
Comment (bkrmendy) [inline editor/src/utils/feature-switches.ts:32]: `TODO` before merge: set to false

File: editor/src/components/editor/actions/actions.tsx [hunk 2]
```diff
@@ -5162,6 +5163,8 @@ export const UPDATE_FNS = {
       imageDragSessionState: action.imageDragSessionState,
     }
   },
+  APPLY_COMMANDS: (action: ApplyCommandsAction, editor: EditorModel): EditorModel =>
+    foldAndApplyCommandsSimple(editor, action.commands),
```
Comment (balazsbajorics) [inline editor/src/components/editor/actions/actions.tsx:5167]: super minor: please use the { return ... } syntax, so it's easier to inject a console.log for debugging etc

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/utils/feature-switches.ts (hunk 1 above)
Comment (enidemi) [reply editor/src/utils/feature-switches.ts:32]: I think a feature switch for 1 extra inspector section is not really necessary

File: editor/src/components/inspector/nine-block-controls.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  const [hovered, setHovered] = React.useState<number>(-1)
+
+  const [flexJustifyContent, flexAlignment] = detectFlexAlignJustifyContent(
+    metadata,
+    selectedViews[0],
+  )
+
+  const setAlignItemsJustifyContent = React.useCallback(
+    (intededFlexAlignment: FlexAlignment, intededJustifyContent: FlexJustifyContent) => {
```
Comment (enidemi) [inline editor/src/components/inspector/nine-block-controls.tsx:71]: typo :)

File: editor/src/components/inspector/nine-block-controls.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  const dispatch = useEditorState((store) => store.dispatch, 'NineBlockControl dispatch')
+  const metadata = useEditorState((store) => store.editor.jsxMetadata, 'NineBlockControl metadata')
+  const selectedViews = useEditorState(
+    (store) => store.editor.selectedViews,
+    'NineBlockControl selectedViews',
+  )
+
+  const [hovered, setHovered] = React.useState<number>(-1)
+
+  const [flexJustifyContent, flexAlignment] = detectFlexAlignJustifyContent(
```
Comment (enidemi) [inline editor/src/components/inspector/nine-block-controls.tsx:65]: please add a TODO to detect if it's set via css only or code or jsx prop

File: editor/src/components/inspector/nine-block-controls.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      const strategies = setFlexAlignJustifyContentStrategies(
+        intededFlexAlignment,
+        intededJustifyContent,
+      )
+      runStrategies(dispatch, metadata, selectedViews, strategies)
+    },
+    [dispatch, metadata, selectedViews],
+  )
+
+  if (selectedViews.length === 0) {
```
Comment (enidemi) [inline editor/src/components/inspector/nine-block-controls.tsx:81]: I think it should be visible only for flex layouts

File: editor/src/components/inspector/nine-block-controls.tsx (hunk 3 above)
Comment (bkrmendy) [reply editor/src/components/inspector/nine-block-controls.tsx:71]: thanks!

**Review by enidemi — APPROVED** [review]

**Pull request merged**